deps/oob-1.0.2/jsonutil.c

```diff
@@ -217,6 +225,124 @@ char *prepareOOBevent(oobOcspData *ocspevent)
   vals = cJSON_CreateObject();
   cJSON_AddItemToObject(list, "Value", vals);
 
+  dns = cJSON_CreateObject();
```

Comment:
Why are you doing this on each call to prepare ?
setoobeventdsninfo should take a map, or an array of pairs or something like that. You should construct this cjson object once. Instead of preparing it repeatedly.

Reply:
There aren't any maps in C and using a struct is the next best thing. This manual creation is unavoidable and already done in the existing code if you look at connectioninfo.

Reply:
That doesn't mean you should do it in each prepare

Reply:
How do you suggest I go about this then? Could you propose an alternative?